From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct snd_emu10k1
Patch-mainline: never, kabi

In 3.12.43, commit 190fa4564d1f2bf7a14d04600c1b9f347af2486c (ALSA:
emu10k1: Emu10k2 32 bit DMA mode), upstream commit
7241ea558c6715501e777396b5fc312c372e11d9 added one member to struct
snd_emu10k1. It indeed changed the layout of that structure.

Since struct snd_emu10k1 is supposed to be allocated by
snd_emu10k1_create, we can just move the member to the end of the
structure and protect it the usual way, by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/sound/emu10k1.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/sound/emu10k1.h
+++ b/include/sound/emu10k1.h
@@ -1709,7 +1709,6 @@ struct snd_emu10k1 {
 	unsigned short model;			/* subsystem id */
 	unsigned int card_type;			/* EMU10K1_CARD_* */
 	unsigned int ecard_ctrl;		/* ecard control bits */
-	unsigned int address_mode;		/* address mode */
 	unsigned long dma_mask;			/* PCI DMA mask */
 	unsigned int delay_pcm_irq;		/* in samples */
 	int max_cache_pages;			/* max memory size / PAGE_SIZE */
@@ -1804,6 +1803,9 @@ struct snd_emu10k1 {
 	bool suspend;
 #endif
 
+#ifndef __GENKSYMS__
+	unsigned int address_mode;		/* address mode */
+#endif
 };
 
 int snd_emu10k1_create(struct snd_card *card,
